File: cielo24_Java/cielo24/src/test/ActionsTest.java
package test;

import org.junit.Before;

import cielo24.Actions;
import cielo24.utils.Guid;

public class ActionsTest {

    protected Config config = new Config();
	protected Actions actions = new Actions(this.config.serverUrl);
    protected Guid apiToken = null;
    protected Guid secureKey = null;

	@Before
	public void setUp() throws Exception {
		if (this.apiToken == null){
			this.apiToken = actions.login(this.config.username, this.config.password, true);
		}
		if (this.secureKey == null){
		    this.secureKey = actions.generateAPIKey(this.apiToken, this.config.username, true);
		}
	}
}

File: cielo24_Ruby/cielo24_gem/test/actions_test.rb
require 'test/unit'
require '../lib/cielo24/actions'
require_relative 'configuration'
include Cielo24

class ActionsTest < Test::Unit::TestCase

  @@config = Configuration.new
  @@actions = Cielo24::Actions.new(@@config.server_url)
  @@api_token = nil
  @@secure_key = nil

  # Called before every test method runs. Can be used
  # to set up fixture information.
  def setup
    if @@api_token.nil?
      @@api_token = @@actions.login(@@config.username, @@config.password, nil, true)
    end
    if @@secure_key.nil?
      @@secure_key = @@actions.generate_api_key(@@api_token, @@config.username, false)
    end
  end
end